src/components/Events/EventFilteredGrid.jsx

```diff
@@ -0,0 +1,232 @@
+import {
+  Box,
+  Grid,
+  GridItem,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Select,
+  Flex,
+} from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { SearchIcon } from '@chakra-ui/icons';
+
+
+import EventCard from '../../components/Events/EventCard';
+import Backend from '../../utils/utils';
+import Fuse from 'fuse.js';
+
+const EventFilteredGrid = () => {
+  const [events, setEvents] = useState([]);
+  const [displayEvents, setDisplayEvents] = useState([]);
+
+  const [selectedEvents, setSelectedEvents] = useState([]);
+  const [name, setName] = useState('');
+  const [dates, setDates] = useState([]);
+  const [locations, setLocations] = useState([]);
+  const [location, setLocation] = useState('');
+  const [date, setDate] = useState('');
+  const [fuse, setFuse] = useState();
+
+  const getEvents = async () => {
+    try {
+      const eventsData = await Backend.get('/events');
+      setEvents(eventsData.data);
+      console.log(eventsData.data);
+      // setDates();
+      setLocations(getLocation(eventsData.data));
+      setDates(getDate(eventsData.data));
+      const options = { keys: ['name', 'date', 'location'], includeScore: true }; //use date and locaiton selected
+      setFuse(new Fuse(eventsData.data, options));
+    } catch (err) {
+      console.log(`Error getting events: `, err.message);
+    }
+  };
+
+  const getLocation = data => {
+    let location = [];
+    for (let i in data) {
+      location.push(data[i].location);
+    }
+    console.log(location.length);
+    return location;
+  };
+
+  const getDate = data => {
+    let date = [];
+    for (let i in data) {
+      date.push(data[i].date.substring(0, 10));
+    }
+    console.log(date.length);
+    return date;
+  };
+
+  const handleCheckboxChange = id => {
+    const newCheckedItems = [...selectedEvents];
+    const index = newCheckedItems.indexOf(id);
+
+    if (index === -1) {
+      newCheckedItems.push(id);
+    } else {
+      newCheckedItems.splice(index, 1);
+    }
+
+    setSelectedEvents(newCheckedItems);
+    console.log(selectedEvents);
+  };
+
+  useEffect(() => {
+    getEvents();
+    console.log(locations);
+    // getEventId(eventId);
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'getEvents' and 'locations'. Either include them or remove the dependency array.